lib/features/email/presentation/widgets/email_attachments_widget.dart

```diff
@@ -0,0 +1,126 @@
+import 'package:core/presentation/extensions/color_extension.dart';
+import 'package:core/presentation/resources/image_paths.dart';
+import 'package:core/presentation/views/button/tmail_button_widget.dart';
+import 'package:filesize/filesize.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:get/get.dart';
+import 'package:model/email/attachment.dart';
+import 'package:model/extensions/list_attachment_extension.dart';
+import 'package:tmail_ui_user/features/base/widget/custom_scroll_behavior.dart';
+import 'package:tmail_ui_user/features/email/presentation/styles/email_attachments_styles.dart';
+import 'package:tmail_ui_user/features/email/presentation/widgets/attachment_file_tile_builder.dart';
+import 'package:tmail_ui_user/main/localizations/app_localizations.dart';
+
+typedef OnDownloadAttachmentActionCallback = Function(BuildContext context, Attachment attachment);
+
+class EmailAttachmentsWidget extends StatefulWidget {
+
+  final List<Attachment> attachments;
+  final OnDownloadAttachmentActionCallback? onDownloadAttachmentActionCallback;
+
+  const EmailAttachmentsWidget({
+    super.key,
+    required this.attachments,
+    this.onDownloadAttachmentActionCallback
+  });
+
+  @override
+  State<EmailAttachmentsWidget> createState() => _EmailAttachmentsWidgetState();
+}
+
+class _EmailAttachmentsWidgetState extends State<EmailAttachmentsWidget> {
+
+  final _imagePaths = Get.find<ImagePaths>();
+
+  bool _isDisplayAll = false;
```

Comment:
you can try `ValueListenerBuilder`

Reply:
The variable `_isDisplayAll ` is used in many sub-widgets of `EmailAttachmentWidget` so we should not use `ValueListenerBuilder ` here.